google/cloud/sql/connector/InstanceConnectionManager.py

```diff
@@ -439,19 +419,145 @@ def immediate_future(object: Any) -> concurrent.futures.Future:
+    def connect(self, driver: str, **kwargs) -> Any:
+        """A method that returns a DB-API connection to the database.
 
-        :rtype: OpenSSl.SSL.Connection
-        :returns: An OpenSSL connection to the primary IP of the database.
+        :type driver: str
+        :param driver: A string representing the driver. e.g. "pg8000" or "pymysql"
+
+        :returns: A DB-API connection to the primary IP of the database.
         """
+        logger.debug("Entered connect method")
+
+        # Host and ssl options come from the certificates and metadata, so we don't
+        # want the user to specify them.
+        kwargs.pop("host", None)
+        kwargs.pop("ssl", None)
+
         with self._lock:
             instance_data: InstanceMetadata = self._current.result()
-        ctx = instance_data.ssl_context
-        ip_addr = instance_data.ip_addresses["PRIMARY"]
-        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        ssl_connection = OpenSSL.SSL.Connection(ctx, s)
-        ssl_connection.connect((ip_addr, 3307))
-        ssl_connection.do_handshake()
-
-        return ssl_connection
+
+        try:
+            connector = {
+                "pymysql": self._connect_with_pymysql,
+                "pg8000": self._connect_with_pg8000,
+            }[driver]
+        except KeyError:
+            raise KeyError("Driver {} is not supported.".format(driver))
+
+        return connector(
+            instance_data.ca_fileobject.name,
+            instance_data.cert_fileobject.name,
+            instance_data.key_fileobject.name,
+            instance_data.ip_address,
+            **kwargs
+        )
+
+    def _connect_with_pymysql(
+        self,
+        ca_filepath: str,
+        cert_filepath: str,
+        key_filepath: str,
+        ip_address: str,
+        **kwargs
+    ):
+        """Helper function to create a pymysql DB-API connection object.
+
+        :type ca_filepath: str
+        :param ca_filepath: A string representing the path to the server's
+            certificate authority.
+
+        :type cert_filepath: str
+        :param cert_filepath: A string representing the path to the ephemeral
+            certificate.
+
+        :type key_filepath: str
+        :param key_filepath: A string representing the path to the private key file.
+
+        :type ip_addresses: Dict[str, str]
+        :param ip_addresses: A Dictionary containing the different IP addresses
+            of the Cloud SQL instance.
+
+        :rtype: pymysql.Connection
+        :returns: A PyMySQL Connection object for the Cloud SQL instance.
+        """
+        try:
+            import pymysql
+        except ImportError:
+            raise ImportError(
+                'Unable to import module "pymysql." Please install and try again.'
+            )
+
+        # ssl_dict = {
+        # "ssl": {"ca": ca_filepath, "cert": cert_filepath, "key": key_filepath, 'check_hostname': False}
```

Comment:
Does this comment need to be here? If so, consider formatting it properly. If not, remove it.